test/util/util.go

```diff
@@ -680,12 +662,22 @@ func ExpectPodsFinalized(ctx context.Context, k8sClient client.Client, keys ...t
 	for _, key := range keys {
 		createdPod := &corev1.Pod{}
 		gomega.EventuallyWithOffset(1, func(g gomega.Gomega) {
-			g.Expect(k8sClient.Get(ctx, key, createdPod)).To(gomega.Succeed())
+			g.Expect(client.IgnoreNotFound(k8sClient.Get(ctx, key, createdPod))).To(gomega.Succeed())
```

Comment:
Hm, but actually as long as the owner workload exists the pod is expected not to be deleted, even if it already does not have the finalizers. So, asserting the set of finalizers (without checking if the pod is gone) seems to make sense.

Reply:
Ah, not the relationship is another way round - pod owns the workload. In that case indeed it might be GC-ed. So, better to check ExpectPodsFinalizedOrGone